apps/pkeyutl.c

```diff
@@ -402,6 +420,15 @@ int pkeyutl_main(int argc, char **argv)
     if (out == NULL)
         goto end;
 
+    if (secoutfile == NULL && pkey_op == EVP_PKEY_OP_ENCAPSULATE) {
+        BIO_printf(bio_err, "Encapsulation requires '-secret' argument\n");
+        goto end;
+    } else {
+        secout = bio_open_default(secoutfile, 'w', FORMAT_BINARY);
+        if (secout == NULL)
+            goto end;
+    }
```

Comment:
Is this needed when the `pkey_op` is `EVP_PKEY_OP_DECAPSULATE`?
Perhaps the `scout` file should not be opened in that case?
The documentation says that the decapsulated secret is written to the `-out` file...

Reply:
I don't think it's crucial but for clarity I changed the code to open this file only for encapsulation